src/Metal/MTLDevice.cs

```diff
@@ -48,6 +48,17 @@ public static partial class MTLDevice {
 		}
 
 #if MONOMAC
+		[Mac (10,11, onlyOn64: true), NoiOS, NoWatch, NoTV]
+		[DllImport (Constants.MetalLibrary)]
+		unsafe static extern IntPtr MTLCopyAllDevices ();
+
+		[Mac (10,11, onlyOn64: true), NoiOS, NoWatch, NoTV]
+		public static IMTLDevice [] GetAllDevices ()
+		{
+			var rv = MTLCopyAllDevices ();
```

Comment:
@spouliot Do you own the return value from `MTLCopyAllDevices`? If so, you'll have to release the handle manually (since NSArray.ArrayFromHandle doesn't do it).